src/components/Homepage.vue

```diff
@@ -25,32 +56,68 @@ import Advertisement from "./Advertisement.vue";
 import { db } from "../db";
 
 export default {
-  name: "homepage",
-  components: {
-    TwitterMsg,
-    TweetDeck,
-    Advertisement
-  },
-
-  data: function() {
-    return {
-      tweets: null,
-      placeholderText: ""
-    };
-  },
-
-  firestore: {
-    tweets: db.collection("batch_one")
-  },
-
-  methods: {
-    // msgReceived(childData) {
-    // 	this.tweets.push(childData);
-    // },
-    // note: Above method was previously used to receive info from TweetDeck
-    updateSomeText(newText) {
-      this.placeholderText = newText;
-    }
-  }
+	name: "homepage",
+	components: {
+		TwitterMsg,
+		TweetDeck,
+		Advertisement
+	},
+
+	data: function() {
+		return {
+			tweets: null,
+			placeholderText: "",
+			names: this.getNames(),
+			display_sortByAuthor: false,
+			tweets_byAuthorName: this.sortByAuthor(name)
+		};
+	},
+
+	firestore: {
+		tweets: db.collection("batch_one")
+	},
+
+	methods: {
+		updateSomeText(newText) {
+			this.placeholderText = newText;
+		},
+		sortByAuthor(name) {
+			if (name === "All") {
+				this.display_sortByAuthor = false;
+			} else {
+				this.display_sortByAuthor = true;
+			}
+
+			var retrievedName = name; // string
+			var tweets_list = [];
+			let self = this;
+			db.collection("batch_one")
+				.where("handle", "==", retrievedName)
+				.get()
+				.then(function(querySnapshot) {
+					querySnapshot.forEach(function(doc) {
+						console.log(doc.id, " => ", doc.data());
```

Comment:
If we want to leave this for debugging, it should indicate what its logging.  For example, if doc.id was undefined for some reason, the console output would be `undefined => undefined` which would be hard to determine.  Suggest adding a prefix.